ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1019,154 @@ def test_server_option_with_unreachable_ad(self):
+        """
+        self.mod_idrange_auto_private_group(type)
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        test_uid = "99999999"
+        test_gid = "98888888"
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        result = self.master.run_command(
+            ["ipa", "idoverrideuser-add", "Default Trust View",
+             "--uid", test_uid, "--gid", test_gid, nonposixuser],
+            raiseonerr=False)
+        if result.returncode != 0:
+            tasks.assert_error(result, "already exists")
+
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.master, nonposixuser)
+        assert(uid == test_uid and gid == test_gid)
+        result = self.clients[0].run_command(
+            ['id', nonposixuser], raiseonerr=False)
+        # Related https://bugzilla.redhat.com/show_bug.cgi?id=2010512
+        tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_nonposixuser_nondefault_primary_group(self, type):
+        """
+        Test for non default primary group
+        """
+        nonposixuser1 = "nonposixuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser1)
+        if type == "true":
+            assert uid == gid
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_posix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust-posix'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_coresponding_existing_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) but there is no group with the
+        corresponding gidNumber.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.master, posixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            assert uid != gid
```

Comment:
You can even check that uid = 10060 (in the 3 cases), and that gid = 10049 if false/hybrid.

Make the same test on the client instead of the master, the user won't be resolvable if false/hybrid.